src/WebJobs.Script/Rpc/Configuration/GenericWorkerProvider.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Azure.WebJobs.Script.Abstractions;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.Rpc
+{
+    internal class GenericWorkerProvider : IWorkerProvider
+    {
+        private WorkerDescription workerDescription;
+        private List<string> arguments;
+
+        public GenericWorkerProvider(WorkerDescription workerDescription, List<string> arguments)
+        {
+            if (this.workerDescription == null)
+            {
+                throw new ArgumentNullException(nameof(workerDescription));
+            }
+
+            if (this.arguments == null)
+            {
+                throw new ArgumentNullException(nameof(arguments));
+            }
+
+            this.workerDescription = workerDescription;
+            this.arguments = arguments;
+        }
+
+        public WorkerDescription GetDescription()
+        {
+            return this.workerDescription;
+        }
+
+        public bool TryConfigureArguments(ArgumentsDescription args, IConfiguration config, ILogger logger)
+        {
+            args.ExecutableArguments.AddRange(arguments);
+            return true;
+        }
+
+        public static List<IWorkerProvider> ReadWorkerProviderFromConfig(ScriptHostConfiguration config, ILogger logger, ScriptSettingsManager settingsManager = null)
+        {
+            var providers = new List<IWorkerProvider>();
+            settingsManager = settingsManager ?? ScriptSettingsManager.Instance;
+            var assemblyDir = Path.GetDirectoryName(new Uri(typeof(WorkerConfigFactory).Assembly.CodeBase).LocalPath);
+
+            var workerDirPath = settingsManager.Configuration.GetSection("workers:config:path").Value ?? Path.Combine(assemblyDir, "workers");
+
+            foreach (var workerDir in Directory.EnumerateDirectories(workerDirPath))
+            {
+                try
+                {
+                    // check if worker config exists
+                    string workerConfigPath = Path.Combine(workerDir, "worker.config.json"); // TODO: Move to constant
+                    if (!File.Exists(workerConfigPath))
```

Comment:
Instead of looking into each directory for config file, look under specific language folders mentioned in config file: "Description":{         "Language":"python" ...}, this will it simpler and enforce existence of config file in the same folder as the worker. Also safer in case lot of directories end up workers dir

Reply:
Not sure I follow? The config file is in the directory that contains the worker. I have to enumerate each directory in the base "worker" directory in order to find `worker.config.json`. It doesn't check recursively.